File: bindings/imgui_bundle/demos_cpp/sandbox/sandbox_plot_in_node.cpp
#include "imgui.h"
#include "imgui-node-editor/imgui_node_editor.h"
#include "implot/implot.h"
#include "immapp/runner.h"
#include "immapp/immapp_widgets.h"
#include <cmath>


namespace ed = ax::NodeEditor;


void gui()
{
    static ImVec2 plotSize(200, 200);
    static double xPos = 5.f;

    auto myPlotFunction = []()
    {
        static std::vector<float> x(1000), y(1000);
        for (int i = 0; i < 1000; ++i) {
            x[i] = i * 0.01f;
            y[i] = sin(x[i]);
        }
        ImPlot::PlotLine("My Line", x.data(), y.data(), 1000);
        ImPlot::DragLineX(0, &xPos, ImVec4(1, 1, 0, 1));
    };

    ed::Begin("My Node Editor");
    ed::BeginNode(ed::NodeId(1));
    ImGui::Text("Hello");
    ImGui::Text("World");
    plotSize = ImmApp::ShowResizablePlotInNodeEditor("My Plot", plotSize, myPlotFunction);
    ed::EndNode();
    ed::End();
}

int main(int, char**)
{
    HelloImGui::RunnerParams runner_params;
    runner_params.callbacks.ShowGui = gui;
    ImmApp::AddOnsParams addons_params;
    addons_params.withImplot = true;
    addons_params.withNodeEditor = true;
    ImmApp::Run(runner_params, addons_params);
}

File: bindings/imgui_bundle/demos_python/sandbox/sandbox_plot_in_node.py
from imgui_bundle import imgui, implot, immapp, ImVec2, ImVec4, imgui_node_editor as ed

x_pos = 5.0
plot_size = ImVec2(20, 10)


def gui():
    global plot_size
    import numpy as np
    x = np.arange(0, 10, 0.01)
    y = np.sin(x)

    def my_plot_function():
        global x_pos
        implot.plot_line("My Line", x, y)
        _, x_pos, _, _, _ = implot.drag_line_x(0, x_pos, ImVec4(1, 1, 0, 1))

    ed.begin("My Node Editor")
    ed.begin_node(ed.NodeId(1))
    imgui.text("Hello")
    imgui.text("World")
    plot_size = immapp.show_resizable_plot_in_node_editor_em("My Plot", plot_size, my_plot_function)
    ed.end_node()
    ed.end()


def main():
    immapp.run(gui, with_implot=True, with_node_editor=True)


if __name__ == "__main__":
    main()
